oem/ibm/libpldmresponder/file_io_type_pcie.cpp

```diff
@@ -1105,5 +1106,24 @@ int PCIeInfoHandler::newFileAvailableWithMetaData(uint64_t /*length*/,
     return PLDM_ERROR_UNSUPPORTED_PLDM_CMD;
 }
 
+void PCIeInfoHandler::deleteTopologyFiles()
+{
+    if (receivedFiles.empty())
+    {
+        try
+        {
+            for (auto& path : fs::directory_iterator(pciePath))
+            {
+                fs::remove_all(path);
+            }
+        }
+        catch (const fs::filesystem_error& err)
+        {
+            std::cerr << "Topology file deletion failed " << pciePath << " : "
+                      << err.what() << "\n";
+        }
+    }
```

Comment:
Do we want an else block on what happens when receivedFiles is not empty? Is this function called only when received files are empty?

Reply:
This function gets called for cable attributes file and topology file. We update received files after getting newFileAvailable for any of these files and we clear received files after receiving file ack for both the files.  So receivedFiles being empty is an indication to tell us that its a refresh(receiving new set of data) and you will get both the files one after the other. Else block is not required here as we need to delete cableinfo and topology file only if its a refresh.